### ISSUE TITLE
Gauge widget - cannot display decimals and units

### ISSUE THREAD

**Issue by rds76** [home-assistant/appdaemon#454]
Gauge shows just rounded number even I specify precision.
I also have to specify units otherwise it shows "undefined", HA has correct unit of measurement set. Btw, basic sensor widget shows it correctly . 
Finally ` units: "&deg;C"` displays: **\&deg;C** expecting: **°C**

**Comment by ReneTode** [comment]
there is no precision setting for gauge, so its obvious that setting it wont work.
its designed for rounded numbers, but precision can be added when we start working on dashboard again.

allthough the docs say that HA units are used i dont see that in the code, so i guess thats missed somehow.

**Comment by IanK6449** [comment]
Hi guys,

It's been a while but has anybody made any progress with this one?
I'd have a crack myself but have no idea how to access the source code for the base widgets. If you can give me any steer in that regard I'll take a look at it.

Cheers, IK

**Comment by IanK6449** [comment]
Yeah I have looked at that and I thought I could create an extended version of the gauge widget from the base version.
I think this is pretty do-able but I am then a bit stuck as to how to modify the code to add a precision parameter.
I was hoping to look at the code for the sensor base widget to get some ideas… but there doesn’t seem to be a base sensor widget.
Consequently I am a bit stuck as to how to proceed but will see if I can do some research on how to do it.
As you’ll probably guess, my JS, CSS and HTML skills are all pretty elementary so fingers crossed.

Thanks for your pointer anyway though.

Cheers,

IK

**Comment by acockburn** [comment]
The sensor widget is derived from the basedisplay widget - the idea being that various types of derived widgets with variations can link back to the same base code to enable some re-use.

**Comment by IanK6449** [comment]
OK, thanks Andrew, that helps.
I have taken a look at the code for the gauge widget and it appears to call a function called JustGage. But I can't find anything on that function itself. Is that some compiled code somewhere in the guts of HADashboard that I won't be able to look at?
Apologies for all the questions!

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by acockburn** [comment]
It’s a 3rd party library included to actually render the gauge graphics.

**Comment by IanK6449** [comment]
OK, thanks Andrew.
Fingers crossed I can get a decimal place working 😄

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by IanK6449** [comment]
Hi Andrew,

Thank you for you assistance / support.

So, I finally managed to get the decimal place working in the gauge by changing the function call to JustGage in the basegauge js file.
I eventually thought to look that function up and discovered the huge list of parameters it can take.
Adding humanReadable true to the function’s parameters seems to have done the trick.
I am not 100% sure this is how that parameter is supposed to work but it does seem to have done what I want so I’m calling that a win 😊
I may try to make that call dependent upon whether precision is specified in the dashboard yaml file next… who knows, it may even work.

**Comment by acockburn** [comment]
Good luck – good progress so far 😊
